### PR TITLE
kubectl version should fail when given extra arguments

### LINKED ISSUE #107955: kubectl version should fail when given extra arguments
ran
```
kubectl version foocluster
```
instead of
```
kubectl version --context foocluster
```
and got no error and then though that was the right version
kubectl should fail for unexpected extra arguments

### PR DESCRIPTION
**Pull request by jlsong01**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
kubectl version should fail when given extra arguments

#### Which issue(s) this PR fixes:

Fixes #107955

#### Does this PR introduce a user-facing change?

```release-note
<kubectl version> fails for unexpected extra arguments
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:77]: Maybe it would be better to wrap this check in a function like in here;

https://github.com/kubernetes/kubernetes/blob/9b09612d1b0a6ec441833f928cbb205d936da59c/staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go#L123

Comment (yxxhero) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:77]: I think so. @jlsong01

Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:77]: fixed

**Comment by jlsong01** [comment]
@ardaguclu @yxxhero PTAL

**Comment by yxxhero** [comment]
@jlsong01 Incomplete unit test, missing verification with args length greater than 0. WDYT?

**Comment by yxxhero** [comment]
@jlsong01  Validate is an export method. If you directly change the function signature, I don't know whether you need API review.

**Comment by yxxhero** [comment]
@jlsong01 I don't know very well. Need to wait for the owner's reply

**Comment by jlsong01** [comment]
@ardaguclu could you take another look please?

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:77]: This block already runs in Validate func?

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:110]: It would be better to use double quote because we don't have anything to escape in that string.

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (ardaguclu) [inline staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:108]: Function signature might be changed according to it https://github.com/kubernetes/kubernetes/blob/9b09612d1b0a6ec441833f928cbb205d936da59c/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L335

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (ardaguclu) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:110]: I think it would be better to return `cmdutil.UsageErr` like in here https://github.com/kubernetes/kubernetes/blob/9b09612d1b0a6ec441833f928cbb205d936da59c/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L335

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:77]: Sorry, I forgot removing it. Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:108]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:110]: Fixed

**Comment by ardaguclu** [comment]
That looks good to me, thanks. But unfortunately I'm not in owners of this PR. Maybe it is better raise your PR on Slack or ping one of sig-cli maintainers.

**Comment by jlsong01** [comment]
ping @dougsland :-)

**Review by dougsland — COMMENTED** [review]
/lgtm

**Comment by jlsong01** [comment]
/assign @soltysh 
for approval

**Comment by sftim** [comment]
Folks, I'd expect to see a changelog entry for this PR.

**Comment by jlsong01** [comment]
> Folks, I'd expect to see a changelog entry for this PR.

Thanks for the comments. The changelog entry has been added into the PR description. PTAL

**Comment by jlsong01** [comment]
@sftim any further comments?

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:104]:
```suggestion
func (o *Options) Validate(args []string) error {
```

We're trying to remove the dependency on cobra directly for these.

**Review by eddiezane — CHANGES_REQUESTED** [review]
Thanks! Small change then lgtm

**Review by eddiezane — APPROVED** [review]
Thanks!

/lgtm
/approve

File: staging/src/k8s.io/kubectl/pkg/cmd/version/version.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:104]: This is wrong place, this is causing tight coupling wit args, `Validate` is meant to ONLY to verify arguments of `Options` struct nothing outside. If you want to introduce such validation this should be done in `Complete` similarly to https://github.com/kubernetes/kubectl/blob/4da03973dd2fcd4645f20ac669d8a73cb017ff39/pkg/cmd/apiresources/apiversions.go#L72-L74

Comment (jlsong01) [reply staging/src/k8s.io/kubectl/pkg/cmd/version/version.go:104]: Thanks for the comments. If no object, I will submit another PR to fix it.
cc @eddiezane

**Pull request merged**
